src/linear_algebra/basis.lean

```diff
@@ -341,6 +341,34 @@ lemma is_basis.smul_eq_zero [no_zero_divisors R] {c : R} {x : M} :
 
 end
 
+lemma is_basis.smul_of_invertible {v : ι → M} (hv : is_basis R v)
+  {w : ι → R} (hw : ∀ i : ι, invertible (w i)) :
```

Comment:
`invertible` is meant to be used by the typeclass system. Wouldn't it be easier to use `is_unit` here?

Reply:
It might also be worth adding a variant for something like
```
lemma is_basis.smul_group {G : Type*} [distrib_mul_action G R] [distrib_mul_action G M]
  [is_scalar_tower G R M] {v : ι → M} (hv : is_basis R v) (w : ι → G) :
```
which after #7438 would suffice to prove the current lemma.

Reply:
I have now added the more general version as `smul_group`